programs/server/config.xml

```diff
@@ -300,16 +299,67 @@
+                    rule - section with matching and mapping info.
+                            Each string will be matched to the regex and, if there is a full match, replaced using format string to
+                             get the name of the local role that needs to be assigned to the user.
+                            There can be multiple 'rule' sections defined inside the same 'role_mapping' section. Each of those rules
+                             will be applied in the order they are listed in 'role_mapping' section. If a rule does not match a string
+                             the next rule will be applied. If none of the rules matched a string and 'fail_if_all_rules_mismatch' is
+                             set to 'false', that particular string will be ignored. If a rule matched a string and 'continue_on_match'
+                             is set to 'false', the subsequent rules will not be applied to the current string.
+                        match - regular expression, in ECMAScript format, used to match each entire string retured by LDAP serach. If
+                             matched successfully, a replacement will be performed and the resulting string will be treated as local
+                             role name and assigned to the user. By default, set to '.+' (match any non-empty string).
+                            Note, that the special characters must be escaped properly in XML.
+                        replace - format string used as a replace expression after the match succeeded. References like '$&' (entire
+                             matched string), or '$n' (n-th subgroup) can be used. By default, set to '$&'.
+                            Note, that the special characters must be escaped properly in XML.
+                        continue_on_match - flag that indicates, whether to continue matching and mapping using the subsequent rules
+                             after this rule successfully matched and mapped the string. By default, set to 'false'.
+                            If set to 'true' and multiple rules match, multiple role names may be generated from one same input string.
              Example:
                 <ldap>
                     <server>my_ldap_server</server>
                     <roles>
                         <my_local_role1 />
                         <my_local_role2 />
                     </roles>
+                    <role_mapping>
+                        <base_dn>ou=groups,dc=example,dc=com</base_dn>
+                        <attribute>cn</attribute>
+                        <scope>subtree</scope>
+                        <search_filter>(&amp;(objectClass=groupOfNames)(member={bind_dn}))</search_filter>
+                        <fail_if_all_rules_mismatch>true</fail_if_all_rules_mismatch>
+                        <rule>
+                            <match>clickhouse_(.+)</match>
```

Comment:
Imagine `i_want_to_hack_clickhouse_admin` group... 

Put anchors! If you did that mistake the users will do even worse mistakes...